org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/LocalCommand.java

```diff
@@ -154,6 +145,224 @@ protected CommandStatus run() {
+				return CommandStatus.FAILED;
+			}
+
+			return CommandStatus.FAILED;
+		}
+
+		// Log the output of the job execution
+		stdOutStream = job.getInputStream();
+		stdErrStream = job.getErrorStream();
+
+		// Check that output was correctly logged. If not, return error
+		if (logOutput(stdOutStream, stdErrStream) == false) {
+			logger.error("Couldn't log output, marking job as failed");
+			return CommandStatus.FAILED;
+		}
+
+		// Try to get the exit value of the job
+		int exitValue = -1; // arbitrary value indicating not completed (yet)
+
+		try {
+			exitValue = job.exitValue();
+		} catch (IllegalThreadStateException e) {
+			// The job is still running, so it should be watched by the
+			// {@link org.eclipse.ice.commands.Command.monitorJob()} function
+
+			logger.info("Job didn't finish, going to monitorJob now");
+			return CommandStatus.RUNNING;
+		}
+		// By convention exit values other than zero mean that the program
+		// failed. If it is not 0, mark the job as failed (since it finished).
+		if (exitValue == 0) {
+			return CommandStatus.SUCCESS;
+		} else {
+			return CommandStatus.FAILED;
+		}
+
+	}
+
+	/**
+	 * See {@link org.eclipse.ice.commands.Command#monitorJob()}
+	 */
+	@Override
+	protected CommandStatus monitorJob() {
+
+		// Local Declarations
+		int exitValue = -1; // Totally arbitrary
+
+		// Wait until the job exits. By convention an exit code of
+		// zero means that the job has succeeded. Watch it until it
+		// finishes.
+		while (exitValue != 0) {
+			// Try to get the exit value of the job
+			// If the job completed successfully this will be 0
+			try {
+				exitValue = job.exitValue();
+			} catch (IllegalThreadStateException e) {
+				// Complain, but keep watching
+				try {
+					commandConfig.getStdErr().write(getClass().getName() + "IllegalThreadStateException!: " + e);
+				} catch (IOException e1) {
+					e1.printStackTrace();
+				}
+			}
+			// Give it a second
+			try {
+				job.waitFor(1000, TimeUnit.MILLISECONDS);
+				// Try again
+				exitValue = job.exitValue();
+			} catch (InterruptedException e) {
+				// Complain
+				try {
+					commandConfig.getStdErr().write(getClass().getName() + " InterruptedException!: " + e);
+				} catch (IOException e1) {
+					e1.printStackTrace();
+				}
+			}
+
+			// If for some reason the job has failed,
+			// it shouldn't be alive and we should break;
+			if (!job.isAlive()) {
+				logger.info("Job is no longer alive, done monitoring");
+				break;
+			}
+		}
+
+		// Print the final exitValue of the job to the output log file
+		try {
+			commandConfig.getStdOut().write("INFO: Command::monitorJob Message: Exit value = " + exitValue + "\n");
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		logger.info("Finished monitoring job with exit value: " + exitValue);
+		if (exitValue == 0)
+			return CommandStatus.SUCCESS;
+		else
+			return CommandStatus.FAILED;
+	}
```

Comment:
Why is an else clause needed? I think it's clearer to use:

if (exitValue != 0) {
	return CommandStatus.FAILED;
}

return CommandStatus.SUCCESS;